python/lsst/pex/logging/loggingLib.py

```diff
@@ -0,0 +1,229 @@
+import lsst.daf.base
+from trace import *
+from _threshold import *
+from _logRecord import *
+from _log import *
+from _debug import *
+from _blockTimingLog import *
+from _screenLog import *
+import _common
+
+import lsst.utils
+
+def getDefaultLog():
+    if _common._DefaultLogIsScreenLog():
+        return _common._getDefaultAsScreenLog();
+    else:
+        return _common.Log_getDefaultLog();
+
+class Prop:
+    """package a property to send it to a LogRecord"""
+    def __init__(self, name, value):
+        self.name = name
+        self.value = value
+
+    def addToRec(self, rec):
+        rec.addProperty(self.name, self.value)
+
+def _LogRecord_addProperty(self, name, val):
+    """add a property with a default type.
+
+    Users can explicitly set the C++ type of a property that gets added by 
+    using the proper type-specific addProperty method (addPropertyBool(), 
+    addPropertyLongLong(), etc.); the methods that are supported this way
+    are int, long, long long, float, double, bool, string, and PropertySet.  
+    This method will choose a type for the property based on the value.  
+
+    If the value is a Python integer, then the value will be set based on 
+    its value:  if the value is out of range of a 32-bit integer, 
+    [-2147483648, 2147483648), it will be stored as a C++ int; out of that 
+    range, the type will be long long.  All floating point numbers are stored
+    as C++ doubles.  Booleans and strings are stored as C++ bools and 
+    std::strings, respectively.  Lists are stored as arrays with the same
+    mappings for the elements (but note that all values in the list must be
+    of the same type).  Dictionaries are stored as PropertySets with similar
+    mappings for their underlying types.
+
+    @param name    the name of the property
+    @param val     that value to set for the property
+    @exception pex.exceptions.TypeError   if the value is of an unsupported type.
+    """
+    if isinstance(val, (int, long)):
+        if val > 2147483648 or val <= -2147483648:
+            return self.addPropertyLongLong(name, val)
+        else:
+            return self.addPropertyInt(name, val)
+    elif isinstance(val, float):
+        return self.addPropertyDouble(name, val)
+    elif isinstance(val, bool):
+        return self.addPropertyBool(name, val)
+    elif isinstance(val, str):
+        return self.addPropertyString(name, val)
+    elif isinstance(val, lsst.daf.base.PropertySet):
+#         return self.addPropertyPropertySetPtr(name, val)
+        raise lsst.pex.exceptions.TypeError("PropertySet type temporarily unsupported")
+    elif isinstance(val, list):
+        for v in val:
+            self.addProperty(name, v)
+    elif isinstance(val, dict):
+        for k in val.keys():
```

Comment:
Indeed, sometimes the python code in the swig files was / is not up to spec. But I decided for clarity to copy it verbatim in most cases. If we indeed go with pybind11 we can clean it up later. If not, then we can still clean it up in the `.i` files with slightly more difficulty.